Add knockdown phrases

gnivler opened the pull request:
Loads from phrases.txt in the mod folder, only when called upon.  Small method to generate the phrases based on settings.  Application of method before calling HandleKnockdown()s

Commit:
refactor to new file and remove Core.cs from changes

Commit:
initial commit - add falling phrases

Commit:
Update CharlesB.cs

janxious commented on Core.cs:
```diff
@@ -12,10 +13,12 @@ public class Core
         public static void Init(string directory, string settingsJSON)
         {
             ModDirectory = directory;
+            KnockdownPhrasePath = Path.Combine(ModDirectory, "phrases.txt");
```
You should move this to after the settings are loaded, and into a helper function inside the `FallingHandling` class.

janxious commented on CharlesB.cs:
```diff
@@ -253,14 +264,56 @@ static bool Prefix(MechMeleeSequence __instance)
-}
\ No newline at end of file
+
+    public static class FallHandling
```
move class to its own file

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	CharlesB.cs
#	Core.cs

gnivler commented:
thanks for the tips - amended my commit.  I think I got away with it in just the one FallHandling class?

janxious commented on CharlesB.cs:
```diff
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
```
not necessary here since you removed the path handling

janxious commented on CharlesB.cs:
```diff
@@ -253,14 +261,16 @@ static bool Prefix(MechMeleeSequence __instance)
                     {
                         target.NeedsInstabilityCheck = true;
-                        target.CheckForInstability();
+                        target.CheckForInstability();                   
```
whitespace

janxious commented on Core.cs:
```diff
@@ -12,7 +13,7 @@ public class Core
         internal static Settings ModSettings = new Settings();
         internal static string ModDirectory;
-
+        
```
whitespace

janxious commented on FallHandling.cs:
```diff
@@ -0,0 +1,45 @@
+{
+    public class FallHandling
+    {
+        private static readonly string KnockdownPhrasePath = Path.Combine(Core.ModDirectory, "phrases.txt");
```
I don't think this is gonna work because ModDirectory is gonna be null when this is assigned. You can make it a lambda `=>` (or a function), but not a variable.

janxious commented on FallHandling.cs:
```diff
@@ -0,0 +1,45 @@
+
+namespace CharlesB
+{
+    public class FallHandling
```
Thinking about naming, I would probably flip the name of the class and the function here to be more like `Piloting.HandleFall.Say()`.

Commit:
refactor to new file and remove Core.cs from changes
file checking for phrases.txt added
ammendments
fix project file references that I mangled and accidentally commited

Commit:
refactor to new file and remove Core.cs from changes
file checking for phrases.txt added

janxious commented on FallHandling.cs:
```diff
@@ -0,0 +1,45 @@
+            var fileLoaded = false; // only initialize the list once, only if it's needed
+            var phrases = new List<string>();
+            var random = new Random();
+            if (!fileLoaded)
```
I feel like you were trying to memoize here, but the the list is not saved to a static variable, so you are loading the file and parsing it every time this method is called.

gnivler commented:
awesome thanks again.  I'll get er

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	HandleFall.cs

Commit:
refactor to new file and remove Core.cs from changes
file checking for phrases.txt added
ammendments
fix project file references that I mangled and accidentally commited
fixed exception handler was wrong

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
refactor to new file and remove Core.cs from changes
file checking for phrases.txt added
ammendments
fix project file references that I mangled and accidentally commited
fixed exception handler was wrong
blasted whitepsaces surviving code cleanup

Commit:
refactor to new file and remove Core.cs from changes
file checking for phrases.txt added
ammendments
fix project file references that I mangled and accidentally commited
fixed exception handler was wrong
add phrases.txt

Commit:
Merge remote-tracking branch 'origin/master'

gnivler commented:
I think I might have it now, look better?

janxious commented:
Can you check in the phrases file?

Commit:
Merge remote-tracking branch 'origin/master'

gnivler commented:
There we go, 15 hand-crafted phrases from my mind and the web.  I took out the really weird ones.

janxious commented:
Check out the last couple commits on this branch https://github.com/janxious/CharlesB/compare/gnivler-master?expand=1

Commit:
refactor to new file and remove Core.cs from changes
file checking for phrases.txt added
ammendments
fix project file references that I mangled and accidentally commited
fixed exception handler was wrong
add phrases.txt - in the wrong place, now moved

gnivler commented:
I hope this is it!  I must have right-clicked Properties when adding the file, didn't notice it went in there.  Fixed memoizing and random class

gnivler commented:
Hi, not sure if there are additional changes requested?  I can't see anything which means little!  Glad to work if stuff is incomplete.

janxious commented:
@gnivler I think it's done. I just haven't made time to test it since the last changes
